controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/DeploymentExpirer.java

```diff
@@ -45,10 +50,19 @@ protected boolean maintain() {
     }
 
     /** Returns whether given deployment has expired according to its TTL */
-    private boolean isExpired(Deployment deployment) {
+    private boolean isExpired(Deployment deployment, ApplicationId instance) {
         if (deployment.zone().environment().isProduction()) return false; // Never expire production deployments
-        return controller().zoneRegistry().getDeploymentTimeToLive(deployment.zone())
-                           .map(timeToLive -> deployment.at().plus(timeToLive).isBefore(controller().clock().instant()))
+
+        Optional<Duration> ttl = controller().zoneRegistry().getDeploymentTimeToLive(deployment.zone());
+        if (ttl.isEmpty()) return false;
+
+        Optional<JobId> jobId = JobType.from(controller().system(), deployment.zone())
+                                       .map(type -> new JobId(instance, type));
+        if (jobId.isEmpty()) return false;
+
+        return controller().jobController().last(jobId.get())
```

Comment:
Could consider a fallback to the deployment's `at` when there is no record of a last job, or when the job-type is unknown. Otherwise LGTM :)

Reply:
Sure, but can that happen?

Reply:
Not now.

Reply:
Ok, if it happens it sounds like a bug somewhere else and I don't think we should work around that here.